workers/unity/Packages/io.improbable.gdk.core/Systems/SpatialOSSendSystem.cs

```diff
@@ -12,19 +16,62 @@ public class SpatialOSSendSystem : ComponentSystem
         private NetworkStatisticsSystem networkStatisticsSystem;
         private NetFrameStats netFrameStats = new NetFrameStats();
 
+        private NativeList<JobHandle> replicationHandles;
+        private List<NativeQueue<SerializedMessagesToSend.UpdateToSend>> componentQueues;
+
+        private ProfilerMarker completingJobsMarker = new ProfilerMarker("SpatialOSSendSystem.CompletingAllJobs");
+        private ProfilerMarker updateQueueMarker = new ProfilerMarker("SpatialOSSendSystem.QueueSerializedMessages");
+
         protected override void OnCreate()
         {
             base.OnCreate();
 
             worker = World.GetExistingSystem<WorkerSystem>();
             networkStatisticsSystem = World.GetOrCreateSystem<NetworkStatisticsSystem>();
+            replicationHandles = new NativeList<JobHandle>(Allocator.Persistent);
+            componentQueues = new List<NativeQueue<SerializedMessagesToSend.UpdateToSend>>();
+        }
+
+        protected override void OnDestroy()
+        {
+            base.OnDestroy();
+            replicationHandles.Dispose();
```

Comment:
Are we potentially leaking memory here with any already serialized data in `componentQueues`?

Reply:
I don't think so. A full frame should always be finished before OnDestroy is called.